### PR TITLE
Write screenshot to file

### PR DESCRIPTION
**Pull request by negiDharmendra**
- [x] Changed default screenshot grabber to write to file and return file path.
- [x] Depricate existing custom screenshot grabber API contract.
- [x] Create a new API contact for custom screenshot grabber to write the screenshot into file and returning file path instead of screenhsot bytes.

```
import java.io.File;
import java.io.IOException;
import java.nio.file.Paths;
import org.apache.commons.io.FileUtils;
import org.openqa.selenium.OutputType;
import org.openqa.selenium.TakesScreenshot;
import com.thoughtworks.gauge.screenshot.CustomScreenshotWriter;

import driver.DriverFactory;

public class CustomScreenshotGrabber implements CustomScreenshotWriter {

    @Override
    public String takeScreenshot() {
        TakesScreenshot driver = (TakesScreenshot) DriverFactory.getDriver();
        File screenshotFile = new File(Paths.get(System.getenv("screenshots_dir"), "screenshot-1.png").toString());
        File tmpFile = driver.getScreenshotAs(OutputType.FILE);
        try {
            FileUtils.copyFile(tmpFile, screenshotFile);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return screenshotFile.getName();
    }

}
```

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
       - name: Install gauge on linux
         if: matrix.os != 'windows-latest'
         run: |
-          git clone https://github.com/getgauge/gauge
+          git clone --depth=1 --branch=plugin-grpc https://github.com/getgauge/gauge
```
Comment (nehashri) [inline .github/workflows/tests.yml]: Please change this to clone from master branch

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -52,20 +52,27 @@ jobs:
...
+
       - name: Install Gauge Java plugin from source
         run: |
           go run make.go
           go run make.go --install
 
       - name: Clone gauge-tests
         run: |
-          git clone --depth=1 https://github.com/getgauge/gauge-tests
+          git clone --depth=1 --branch=fileBasedScreenshot https://github.com/getgauge/gauge-tests
```
Comment (nehashri) [inline .github/workflows/tests.yml]: The tests should also be cloned from master branch

File: src/main/java/com/thoughtworks/gauge/screenshot/ICustomScreenshot.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.thoughtworks.gauge.screenshot;
+
+public interface ICustomScreenshot {
```
Comment (nehashri) [inline src/main/java/com/thoughtworks/gauge/screenshot/ICustomScreenshot.java]: This is not java syntax. The use of `I` before an interface is in C# and not Java. Please change this to CustomScreenshot

File: src/main/java/com/thoughtworks/gauge/screenshot/ICustomScreenshotWriter.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2015 ThoughtWorks, Inc.
```
Comment (nehashri) [inline src/main/java/com/thoughtworks/gauge/screenshot/ICustomScreenshotWriter.java]: copyright should be 2019

File: src/main/java/com/thoughtworks/gauge/screenshot/ICustomScreenshotWriter.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+package com.thoughtworks.gauge.screenshot;
+
+/**
+ * Implement this interface to take Custom Screenshots on Failure. It overrides the default screenshot taking mechanism.
+ * The captured screenshots can be seen on the reports on failure.
+ * If multiple implementations are found, one will be picked randomly to capture screenshots.
+ * Implementation of this interface should capture screenshot and write them into a unique file
+ * inside screenshots directory. Use "screenshots_dir" env to get screenshot directory path"
+ */
+public interface ICustomScreenshotWriter extends ICustomScreenshot {
```
Comment (nehashri) [inline src/main/java/com/thoughtworks/gauge/screenshot/ICustomScreenshotWriter.java]: This is not java syntax. The use of I before an interface is in C# and not Java. Please change this to CustomScreenshotWriter

File: src/main/java/com/thoughtworks/gauge/GaugeConstant.java [hunk 6]
```diff
@@ -23,4 +23,5 @@
     public static final String GAUGE_PROJECT_ROOT = "GAUGE_PROJECT_ROOT";
     public static final String GAUGE_CUSTOM_COMPILE_DIR = "gauge_custom_compile_dir";
     public static final String DEFAULT_SRC_DIR = "src/test/java";
+    public static final String SCREENSHOTS_DIR = "screenshots_dir";
```
Comment (nehashri) [inline src/main/java/com/thoughtworks/gauge/GaugeConstant.java]: I think this is the name of the env. Please rename to `SCREENSHOTS_DIR_ENV`

File: .github/workflows/tests.yml (hunk 2 above)
Comment (nehashri) [reply .github/workflows/tests.yml]: @negiDharmendra the tests are still cloned from a file.

Comment (negiDharmendra) [reply .github/workflows/tests.yml]: We can not clone from the `master`branch as there are some changes related to html-report FTs in the `fileBasedScreenshot` branch. Those changes can not be pushed to master until this [issue](https://github.com/getgauge/gauge/issues/1476) is addressed by all the runner.

**Pull request merged**